server/server_test.go

```diff
@@ -3295,15 +3303,62 @@ func TestDontEmbedNATSNotRunning(t *testing.T) {
 	failedServer = RunServerWithOpts(sOpts, nil)
 }
 
-func TestDontEmbedNATRunning(t *testing.T) {
+func TestDontEmbedNATSRunning(t *testing.T) {
 	sOpts := GetDefaultOptions()
-	sOpts.EmbedNATS = false
+	sOpts.NATSServerURL = "nats://localhost:5223"
 
 	nOpts := DefaultNatsServerOptions
 	nOpts.Host = "localhost"
 	nOpts.Port = 5223
-	natsdTest.RunServer(&nOpts)
+	natsd := natsdTest.RunServer(&nOpts)
+	defer natsd.Shutdown()
 
 	s := RunServerWithOpts(sOpts, &nOpts)
 	defer s.Shutdown()
 }
+
+func TestDontEmbedNATSMultipleURLs(t *testing.T) {
+	nOpts := DefaultNatsServerOptions
+	nOpts.Host = "localhost"
+	nOpts.Port = 5223
+	nOpts.Username = "ivan"
+	nOpts.Password = "pwd"
+	natsServer := natsdTest.RunServer(&nOpts)
+	defer natsServer.Shutdown()
+	auth := &auth.Plain{Username: nOpts.Username, Password: nOpts.Password}
+	natsServer.SetClientAuthMethod(auth)
+
+	sOpts := GetDefaultOptions()
+
+	workingURLs := []string{
+		"nats://localhost:5223",
+		"nats://ivan:pwd@localhost:5223",
+		"nats://ivan:pwd@localhost:5223, nats://ivan:pwd@localhost:5224",
+	}
+	for _, url := range workingURLs {
+		sOpts.NATSServerURL = url
+		s := RunServerWithOpts(sOpts, &nOpts)
+		s.Shutdown()
+	}
+
+	notWorkingURLs := []string{
+		"nats://ivan:incorrect@localhost:5223",
+		"nats://localhost:5223,nats://ivan:pwd@localhost:5224",
+		"nats://localhost",
+		"localhost:5223",
+		"localhost",
```

Comment:
Perhaps a few more to cover bases parsing, `"nats://ivan:pwd@:5224"`, `""`, and `"  "`.